### PR TITLE
Added doc for absolute import from gatsby eslint rules or other tools

### LINKED ISSUE #22194: docs: absolue imports
## Summary
Currently the [docs](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/add-custom-webpack-config.md) recommend to set absolute imports within webpack. While this does work it only applies to webpack. If you are trying to eject from gatsby's eslint rules or use other tools they will not know how to resolve these and will complain. For those using yarn 1 or 2 there is a better way:

the best practice is to set them in your package.json:

```json
{
  "dependencies": {
      "hooks": "link:./src/hooks",
  }
}
```

I read this some where more authoritative but can't find where (yarn docs I believe), I'll keep looking and update this if I find the link.

https://yarnpkg.com/features/protocols

### Motivation

This allows all tools to resolve these paths and preps you for yarn pnp.

### Draft the doc

- [ ] Update the doc, following the format listed in these resources:
  - [Overview on contributing to documentation](https://www.gatsbyjs.org/contributing/docs-contributions/)
  - [Docs Templates](https://www.gatsbyjs.org/contributing/docs-templates/)
  - [Example of a similar article]()

### Open a pull request

- [ ] Open a pull request with your work including the words "closes #[this issue's number]" in the pull request description

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #22194 .  
@moonmeister , Please review.

### CODE REVIEW

**Comment by laurieontech** [comment]
Thanks for the PR! However, we're in a discussion about where this information should exist. As this is a doc specifically about webpack, it likely doesn't belong here. See [my comment on the issue](https://github.com/gatsbyjs/gatsby/issues/22194#issuecomment-606183071).

I appreciate that you picked up an issue from a community member and apologize that you put in the work prior to us determining the best course forward.

I'm going to close this for the moment. That being said, we may re-open it and move the text you provided elsewhere. Thank you again for contributing.

**Comment by laurieontech** [comment]
I'm going to re-open this and see if we can adjust it to account for this comment: https://github.com/gatsbyjs/gatsby/issues/22194#issuecomment-610244233

The goal here is to give the link and information for npm and yarn. Using note notation should help (> in markdown).

Let me know if you're able to make these adjustments. Thanks so much!

**Comment by ashutosh1919** [comment]
@laurieontech , @moonmeister and @wardpeet , I have created new section for this. Please check it and suggest changes if required. And please review this PR.

File: docs/docs/add-custom-webpack-config.md [hunk 1]
```diff
@@ -75,6 +75,20 @@ exports.onCreateWebpackConfig = ({ stage, actions }) => {
 
 You can always find more information on _resolve_ and other options in the official [Webpack docs](https://webpack.js.org/concepts/).
 
+### Importing non-webpack tools using `yarn`
+
+Note that above solution absolute imports only applies to webpack. If you are trying to
```
Comment (laurieontech) [inline docs/docs/add-custom-webpack-config.md:80]:
```suggestion
Note that using absolute imports only applies to webpack resolutions and will not work for other tools, e.g. eslint or typescript. 
```

File: docs/docs/add-custom-webpack-config.md [hunk 2]
```diff
@@ -75,6 +75,20 @@ exports.onCreateWebpackConfig = ({ stage, actions }) => {
...
+eject from gatsby's eslint rules or use other tools they will not know how to resolve
```
Comment (laurieontech) [inline docs/docs/add-custom-webpack-config.md:81]:
```suggestion
```

File: docs/docs/add-custom-webpack-config.md [hunk 3]
```diff
@@ -75,6 +75,20 @@ exports.onCreateWebpackConfig = ({ stage, actions }) => {
...
+these and will complain. But if you are using yarn 1 or 2, then the best practice is to set them in package.json as shown below:
```
Comment (laurieontech) [inline docs/docs/add-custom-webpack-config.md:82]:
```suggestion
But if you are using yarn, then the best practice is to set up your imports in package.json as shown below:
```

**Comment by ashutosh1919** [comment]
@laurieontech , I think in my opinion, we should keep this with heading (searchable). It will work for both the cases which you described.

**Comment by laurieontech** [comment]
I'm ok with that. Let's get this merged.

**Pull request merged**